webdriver/tests/bidi/browsing_context/navigate/navigate.py

```diff
@@ -108,3 +109,44 @@ async def test_same_document_navigation_in_before_unload(bidi_session, new_tab,
 
     url_after = url_before.replace("empty.html", "other.html")
     await navigate_and_assert(bidi_session, new_tab, url_after, "complete")
+
+
+@pytest.mark.parametrize(
+    "script",
+    [
+        "<script>window.location='{url}'</script>",
+        """<script>window.addEventListener('DOMContentLoaded', () => {{
+            window.location = '{url}';
+        }});</script>""",
+        """<script>window.addEventListener('load', () => {{
+            window.location = '{url}';
+       }});</script>""",
+    ],
+    ids=["Redirect immediately", "Redirect on DOMContentLoaded", "Redirect on load"],
```

Comment:
I don't think `redirect` is the right term here. Perhaps it's `replaced` or `interrupted`.

Reply:
Updated